examples/api/test/painting/linear_gradient.0_test.dart

```diff
@@ -0,0 +1,39 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_api_samples/painting/gradient/linear_gradient.0.dart'
+    as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('finds a gradient', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      const MaterialApp(
+        home: example.MoodyGradient(),
+      ),
+    );
+
+    expect(find.byType(example.MoodyGradient), findsOneWidget);
+  });
+
+  testWidgets('gradient matches golden', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      const MaterialApp(
+        home: SizedBox(
+          width: 800,
+          height: 600,
+          child: RepaintBoundary(
+            child: example.MoodyGradient(),
+          ),
+        ),
+      ),
+    );
+    await expectLater(
+      find.byType(example.MoodyGradient),
+      matchesGoldenFile('linear_gradient.0_test.png'),
```

Comment:
Can you include `api` in the name so it can be distinguished on the Gold dashboard? 
Maybe

```suggestion
      matchesGoldenFile('api.linear_gradient.0_test.png'),
```

Reply:
Ideally, we would configure this prefix globally in something like `flutter_test_config.dart`, so not everybody writing a test has to remember to put this here. Do you think that would be doable?

Reply:
I think so... I am guessing it would be best to add in the same place we added an assertion in https://github.com/flutter/flutter/pull/99082
If the path contains `api`, `_addPrefix` can include `api` in the prefix. Do you want to add that here, or I can in a separate change.

Reply:
Done.